### PR TITLE
Bugfix/s3 utils 27 throttle calls to sproxyd

### PR DESCRIPTION
**Pull request by watsaqat**

### CODE REVIEW

File: ObjectRepair/SproxydKeysSubscribers.js [hunk 1]
```diff
@@ -46,13 +48,24 @@ class DuplicateSproxydKeyFoundHandler {
             objectUrl2,
         };
 
-        return this._repairObject(objInfo, err => {
+
+        this.queue.push(objInfo, err => {
```
Comment (watsaqat) [inline ObjectRepair/SproxydKeysSubscribers.js:52]: push onto a worker queue to control concurrency

**Comment by watsaqat** [comment]
In addition to throttling requests, this PR also improves logging when an object is deleted before repair and refactors naming conventions to reflect that this service will be called Object Repair going forward

**Review by tmacro — APPROVED** [review]

File: ObjectRepair/DuplicateKeysIngestion.js [hunk 2]
```diff
@@ -27,14 +27,15 @@ class RaftJournalReader {
      * @param {number} sessionId - Raft Session Id that the reader will poll from.
      */
     constructor(begin, limit, sessionId) {
-        this.lookBack = Number.parseInt(LOOKBACK_WINDOW, 10);
+        this.lookBack = Number.parseInt(OBJECT_REPAIR_LOOKBACK_WINDOW, 10);
+        this.aggregateLogger = new AggregateLogger(env.OBJECT_REPAIR_LOG_INTERVAL);
```
Comment (jonathan-gramain) [inline ObjectRepair/DuplicateKeysIngestion.js:25]: minor: could be extracted from the env above, along with the other variables

File: ObjectRepair/DuplicateKeysWindow.js [hunk 3]
```diff
@@ -87,7 +87,7 @@ class SproxydKeysProcessor {
     checkDuplicate(sproxydKey, objectKey, bucket) {
         const existingObjectKey = this.sproxydKeys.get(sproxydKey);
         if (existingObjectKey && existingObjectKey !== objectKey) {
-            log.warn('object keys with a duplicate sproxyd key found', {
+            log.info('object keys with a duplicate sproxyd key found', {
```
Comment (jonathan-gramain) [inline ObjectRepair/DuplicateKeysWindow.js]: I'd rather leave this as a warning log (abnormal situation, raising an alert)

File: ObjectRepair/Logging.js [hunk 4]
```diff
@@ -13,7 +20,7 @@ class AggregateLogger {
      * @param {number} logInterval - log summary statistics once every intervalSize seconds
      */
     constructor(logInterval) {
-        this.log = new werelogs.Logger('s3utils:SproxydKeysScan:AggregateLogger');
+        this.log = new werelogs.Logger('ObjectRepair:SproxydKeysScan:AggregateLogger');
```
Comment (jonathan-gramain) [inline ObjectRepair/Logging.js]:
```suggestion
        this.log = new werelogs.Logger('ObjectRepair:AggregateLogger');
```

File: ObjectRepair/Logging.js [hunk 5]
```diff
@@ -49,7 +57,7 @@ class AggregateLogger {
             endTime: currentTime.toISOString(),
             summary,
         };
-        this.log.info('Summary update:', data);
+        this.log.warn('Summary update:', data);
```
Comment (jonathan-gramain) [inline ObjectRepair/Logging.js:60]: should stay as info log?

File: ObjectRepair/index.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+Usage:
+    node ObjectRepair
+
+Mandatory environment variables:
+    OBJECT_REPAIR_BUCKETD_HOSTPORT: <bucketd_host>:<bucketd_port>
+    OBJECT_REPAIR_SPROXYD_HOSTPORT: <sproxyd_host>:<sproxyd_port>
+    OBJECT_REPAIR_RAFT_SESSION_ID: Session id from which to read Journal.
+    OBJECT_REPAIR_RAFT_LOG_BATCH_SIZE: Number of records to fetch at each poll of the Raft Journal.
+    OBJECT_REPAIR_LOOKBACK_WINDOW: When the process is started/restarted it will begin at 
+        cseq - OBJECT_REPAIR_LOOKBACK_WINDOW unless OBJECT_REPAIR_RAFT_LOG_BEGIN_SEQ is set explicitly.
```
Comment (jonathan-gramain) [inline ObjectRepair/index.js]: I believe those two environment variables are optional

**Review by jonathan-gramain — APPROVED** [review]
some minor comments, overall LGTM

**Pull request merged**